### PR TITLE
Artifact events

### PR DESCRIPTION
**Pull request by t-persson**
### Description of the Change

Publish Artifact and Report events on the internal messagebus when uploading files.

### Alternate Designs

None!

### Possible Drawbacks

None!

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

File: src/etos_test_runner/lib/log_area.py [hunk 1]
```diff
@@ -253,9 +278,12 @@ def __upload(
         if upload.get("auth"):
             upload["auth"] = self.__auth(**upload["auth"])
 
+        checksums = {}
         with open(log, "rb") as log_file:
+            content = log_file.read()
+            checksums["md5"] = hashlib.md5(content).hexdigest()
```
Comment (t-persson) [inline src/etos_test_runner/lib/log_area.py:284]: We should probably another algorithm. Eiffel uses these: `SHA-224, SHA-256, SHA-384, SHA-512, SHA-512/224, SHA-512/256`

File: src/etos_test_runner/lib/log_area.py [hunk 2]
```diff
@@ -282,8 +311,8 @@ def __retry_upload(
         :type verb: str
         :param url: URL to retry upload request
         :type url: str
-        :param log_file: Opened log file to upload.
-        :type log_file: file
+        :param file_contents: File contents to upload
+        :type file_contents: str
```
Comment (andmat900) [inline src/etos_test_runner/lib/log_area.py:315]: the file is read as bytes, not str

File: src/etos_test_runner/lib/events.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        self.publisher = publisher
+        self.identifier = etos.config.get("suite_id")
+
+    def __del__(self):
+        """Close the RabbitMQ publisher."""
+        self.close()
+
+    def close(self):
+        """Close the RabbitMQ publisher if it is started."""
+        if self.publisher is not None and self.publisher.is_alive():
```
Comment (andmat900) [inline src/etos_test_runner/lib/events.py:48]: log a message if any of this fails?

**Review by fredjn — DISMISSED** [review]

File: src/etos_test_runner/lib/events.py (hunk 3 above)
Comment (t-persson) [reply src/etos_test_runner/lib/events.py:48]: There should be enough logging from eiffel-pythonlib for RabbitMQ errors. It is quite verbose so adding more logs would be very noisy.

**Comment by t-persson** [comment]
Force pushed as I had to rebase. The only change is updating "str" to "bytes" in a docstring. Will merge if checks pass.

**Pull request merged**